core/src/smtp/mod.rs

```diff
@@ -69,6 +69,11 @@ pub async fn check_smtp(
 			.await
 			.map_err(|err| err.into());
 	}
+	if input.outlook_use_api && host_lowercase.ends_with(".outlook.com.") {
```

Comment:
Hotmail addresses seem to resolve to `hotmail-com.olc.protection.outlook.com.` so they'll get caught by this clause. However, it will result in an error as the OneDrive trick doesn't work for that provider.

All Office365-backed domains, however, seem to resolve to `<domain>.mail.protection.outlook.com.` so other than specifically excluding the above `hotmail-com…` host, I'm not sure if there's an easy distinction.

Reply:
I tested a couple of emails:
- @outlook.com -> `outlook-com.olc.protection.outlook.com.`
- @outlook.fr -> `eur.olc.protection.outlook.com.`
- @hotmail.com -> `hotmail-com.olc.protection.outlook.com.` 
- @hotmail.fr -> `eur.olc.protection.outlook.com.`
- @domain.tld -> `domain-tld.mail.protection.outlook.com.` (tested with 2 different companies)

So it seems to me that we can do:
```suggestion
	if input.outlook_use_api && host_lowercase.ends_with(".mail.protection.outlook.com.") {
```